src/pages/background/index.ts

```diff
@@ -9,4 +11,27 @@ reloadOnUpdate('pages/background');
  */
 reloadOnUpdate('pages/content/style.scss');
 
-console.log('background loaded');
+const calcTsuriScore = async (videoId: string): Promise<number> => {
+  const tsuriScoreAPIResponse = await tsuriScoreAPI({ video_ids: [videoId] });
+  const tsuriScore = tsuriScoreAPIResponse.items[0].tsuri_score;
+  return tsuriScore;
+};
+
+chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
+  const message = request as Message;
+
+  if (message.type === 'TsuriScoreRequest') {
+    (async () => {
+      const tsuriScoreRequest = message as TsuriScoreRequestMessage;
+      const tsuriScore = await calcTsuriScore(tsuriScoreRequest.video_id);
+      const response: TsuriScoreResponseMessage = {
+        type: 'TsuriScoreResponse',
+        video_id: tsuriScoreRequest.video_id,
+        tsuri_score: tsuriScore,
+      };
+      sendResponse(response);
+    })();
+    // To indicate that it will respond asynchronously
```

Comment:
> Caution: If you use callbacks, the sendResponse() callback is only valid if used synchronously, or if the event handler returns true to indicate that it will respond asynchronously. The sendMessage() function's callback is automatically invoked if no handlers return true or if the sendResponse() callback is garbage-collected.

https://developer.chrome.com/docs/extensions/develop/concepts/messaging#simple

非同期関数内では`return true`しないと常に`undefined`をreturnされる．